### ISSUE TITLE
bmr plot broken

### ISSUE THREAD

**Issue by jakob-r** [mlr-org/mlr3viz#62]
```r
library(mlr3)
library(mlr3viz)

# benchmarking with benchmark_grid()
design = benchmark_grid(tsk("iris"), lrn("classif.rpart"), rsmp("cv", folds = 3))
bmr = benchmark(design)

design2 = benchmark_grid(tsk("sonar"), lrn("classif.featureless"), rsmp("cv", folds = 3))
bmr2 = benchmark(design2)

bmr$combine(bmr2)
plot(bmr)
```
![image](https://user-images.githubusercontent.com/1888623/105995505-5cf17680-60a9-11eb-8fb5-2a7c4c255325.png)

This also happens in more realistic scenarios.

I tried to fix it here:
https://github.com/mlr-org/mlr3viz/blob/777fb80f8e6b6d07785dab1ae9ecee4d1f71e6a5/R/BenchmarkResult.R#L57
by adding `names(learner_labels) = unique(tab$nr)` whihc would work for the given example but the test fails.
The test fail indicates that there is no bijective mapping between `tab$nr` and `learner_labels`. If there is no such mapping it points to a further problem that the labels cannot be correct.

**Comment by mllg** [comment]
I'm not sure if I understand the problem. Why is `tab$nr` mapped to aesthetic x, only to overwrite the labels with `scale_x_discrete()`?

Won't this work?
```r
      ggplot(tab, mapping = aes(x = .data$learner_id, y = .data[[measure_id]])) +
        geom_boxplot() +
        labs(x = "") +
        facet_wrap(vars(.data$task_id), scales = "free_x")

```

**Comment by jakob-r** [comment]
I guess the answer is no as a simple blame reveals: https://github.com/mlr-org/mlr3viz/commit/86f5f129812decde5b87442ed731778182e096fe

**Comment by mllg** [comment]
Solved via #63.
